src/proposal.jl

```diff
@@ -21,21 +21,22 @@ abstract type DynamicTrajectory{I<:AbstractIntegrator} <: AbstractHamiltonianTra
 abstract type NoUTurnTrajectory{I<:AbstractIntegrator} <: DynamicTrajectory{I} end
 struct NUTS{I<:AbstractIntegrator} <: NoUTurnTrajectory{I}
     integrator  ::  I
+    max_depth   ::  Int
+    Δ_max       ::  AbstractFloat
 end
 
+# Helper function to use default values
+NUTS(integrator::AbstractIntegrator) = NUTS(integrator, 10, 1000.0)
+
 # Create a `NUTS` with a new `ϵ`
 function (snuts::NUTS)(ϵ::AbstractFloat)
```

Comment:
This helper's argument is potentially confusing since both `Δ_max` and `ϵ` can be float numbers. What if you want to create a new `NUTS` algorithm, with a different `Δ_max`? Perhaps consider the following

```julia
function (snuts::NUTS)(i::AbstractIntegrator)
   return NUTS(i, snuts.max_depth, snuts.Δ_max)
end
```

as an alternative

Reply:
I will change as suggested. But we will probably change this helper interface (https://github.com/TuringLang/AdvancedHMC.jl/issues/45).